spec.bs

```diff
@@ -556,6 +644,9 @@ A [=user agent=] holds an <dfn>aggregation coordinator map</dfn>, which is a
 A [=user agent=] holds a <dfn>pre-specified report parameters map</dfn>, which
 is a [=map=] from [=batching scopes=] to [=pre-specified report parameters=].
 
+A [=user agent=] holds a <dfn>named budget ledgers map</dfn>, which is a [=map=]
+from [=batching scopes=] to [=lists=] of [=named budget ledgers=].
+
```

Comment:
I realized I omitted a key part, which is that budget consumption against named budgets is persisted on disk and that implementations will set practical limits, like "no more than 25 distinct named budgets over 24 hours".

It's not clear to me what the "lifetime" of the user agent is. Can we model storing to the disk just by updating these structures held by the user agent? If so, maybe I should add a new "persistent storage" object held by the user agent and explicitly store records of budget consumption per minute.

Reply:
Gotcha. Still mulling on this, but in the meantime I left an issue about moving the data away from the user agent and onto the {{PrivateAggregation}} object. But yeah, we might be able to avoid the field entirely?

Reply:
One other issue that comes to mind here is that for Protected Audience "batching scopes" are auctions, but the explainer says reservations are scoped to a single worklet/script runner context. Should we be using "debug scopes" here? 

One other complication with moving reservations onto the {{PrivateAggregation}} object is handling contributions triggered by fenced frames via `window.fence.reportEvent()` (as those contributions outlive the auction).